File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
           int first = findBound(nums, target, true);
        if (first == -1) {
            return new int[]{-1, -1};
        }
        int last = findBound(nums, target, false);
        return new int[]{first, last};
    }

    static int findBound(int[] nums, int target, boolean isFirst) {
        int start = 0;
        int end = nums.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (nums[mid] == target) {
                if (isFirst) {
                    if (mid == start || nums[mid - 1] != target) {
                        return mid; // Found first occurrence
                    }
                    end = mid - 1; // Search left
                } else {
                    if (mid == end || nums[mid + 1] != target) {
                        return mid; // Found last occurrence
                    }
                    start = mid + 1; // Search right
                }
            } else if (nums[mid] > target) {
                end = mid - 1; // Target is on the left side
            } else {
                start = mid + 1; // Target is on the right side
            }
        }
        return -1; // Target not found
    }
}

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var binary=(nums,target,isSerach)=>{
     let start=0;
    let end=nums.length-1
    let ans=-1
    while(start<=end){
        let mid=Math.floor((start+end)/2)
        if(nums[mid]===target){
            ans=mid
            if(isSerach){
                end=mid-1
            }else{
                start=mid+1
           
        }}else if(nums[mid]<target)
            start=mid+1
        else
            end=mid-1
        
        
    }
    return ans


}
var searchRange = function(nums, target) {
   let start=binary(nums,target,true)
let end=binary(nums,target,false)
return  [start,end]



};